### DISCUSSION TITLE
Альтернативные транспортные протоколы

### DISCUSSION THREAD

**Discussion by S-trace** [xtrime-ru/antizapret-vpn-docker#18] [Ideas]
OpenVPN уже давненько не самое ркностойкое решение.
Возможно, есть смысл подумать об альтернативном транспорте?

Судя по архитектуре решения, добавление альтернативных транспортов не представляет сложности - antizapret предоставляет данные, kres резолвит, ferm добавляет правила DNAT, OVPN рулит подключениями, можно добавить паралельно ему альтернативные транспорты типа awg. shadowsocks или VLESS.

Только надо решить вопрос как перехватывать весь (!) DNS изо всей сети (вы же не хотите настраивать все девайсы индивидуально?)) и направлять его в сторону сервера  (помним про DoH, DoT и DNSSec). 
И это (скорее всего) более-менее платформоспецифичный вопрос, так как в разных транспортах на разных платформах могут быть самые причудливые возможности и ограничения.

Давайте это обсудим?

**Comment by xtrime-ru** [comment]
Да, это напрашивается. Но хотелось бы не пихать все в один контейнер, а поднимать отдельные под каждый протокол и потом перенапрявлять все пакеты через iptables в контейнер антизапрета. 

Тогда можно будет брать готовые контейнеры и интегрировать их минимальными усилиями.

**Comment by xtrime-ru** [comment]
@S-trace Готово: https://github.com/xtrime-ru/antizapret-vpn-docker/pull/49

Нет желания какой нибудь openconnect по аналогии прикрутить?
